### DISCUSSION TITLE
RejectOnNotFound Error Catching

### DISCUSSION THREAD

**Discussion by jgee67** [prisma/prisma#9768] [Q&A]
How does one successfully catch the default error thrown from the RejectOnNotFound setting? It would be nice if this error had a class name to go with it, but I'm not finding anything exported for it. I can just catch Error and make a conditional if `error.name === "NotFoundError"`, but that feels sloppy

**Comment by ryands17** [comment]
@jgee67 👋 

Unfortunately string comparison is the only way to handle this as of now. We have an epic descried here that will be focused on improving this: https://github.com/prisma/prisma/issues/3488

In the meantime, could you try with a [custom error](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#rejectonnotfound) in this case?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
